test/e2e/suite/cases/audit/audit.go

```diff
@@ -0,0 +1,269 @@
+    certificate-authority: /audit-webhook-ca/ca.pem
+  name: logstash
+contexts:
+- context:
+    cluster: logstash
+    user: ""
+  name: default-context
+current-context: default-context
+preferences: {}
+users: []`,
+			},
+		})
+		Expect(err).NotTo(HaveOccurred())
+
+		vols := []corev1.Volume{
+			corev1.Volume{
+				Name: "audit",
+				VolumeSource: corev1.VolumeSource{
+					ConfigMap: &corev1.ConfigMapVolumeSource{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: cmPolicy.Name,
+						},
+					},
+				},
+			},
+			corev1.Volume{
+				Name: "audit-webhook",
+				VolumeSource: corev1.VolumeSource{
+					ConfigMap: &corev1.ConfigMapVolumeSource{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: cmWebhook.Name,
+						},
+					},
+				},
+			},
+			extraWebhookVol,
+		}
+
+		By("Deploying proxy with audit policy enabled")
+		f.DeployProxyWith(vols, "--audit-webhook-config-file=/audit-webhook/kubeconfig.yaml",
+			"--audit-policy-file=/audit/audit.yaml", "--audit-webhook-initial-backoff=1s", "--audit-webhook-batch-max-wait=1s")
+
+		testAuditLogs(f, "app=audit-webhook-e2e")
+	})
+})
+
+func testAuditLogs(f *framework.Framework, podLabelSelector string) {
+	By("Making calls to proxy to ensure audit get created")
+	token := f.Helper().NewTokenPayload(f.IssuerURL(), f.ClientID(), time.Now().Add(time.Second*5))
+	signedToken, err := f.Helper().SignToken(f.IssuerKeyBundle(), token)
+	Expect(err).NotTo(HaveOccurred())
+
+	proxyConfig := f.NewProxyRestConfig()
+	requester := f.Helper().NewRequester(proxyConfig.Transport, signedToken)
+
+	target := fmt.Sprintf("%s/api/v1/namespaces/kube-system/pods", proxyConfig.Host)
+
+	// Make request that should succeed
+	_, _, err = requester.Get(target)
+	Expect(err).NotTo(HaveOccurred())
+
+	// Make request that should be unauthenticated
+	requester = f.Helper().NewRequester(proxyConfig.Transport, "foo")
+	_, resp, err := requester.Get(target)
+	Expect(err).NotTo(HaveOccurred())
+
+	if resp.StatusCode != http.StatusUnauthorized {
+		Expect(fmt.Errorf("expected to get unauthorized, got=%d", resp.StatusCode)).NotTo(HaveOccurred())
+	}
+
+	By("Waiting for audit logs to be written")
+	time.Sleep(time.Second * 5)
+
+	By("Copying audit log from proxy locally")
+	// Get pod
+	pods, err := f.Helper().KubeClient.CoreV1().Pods(f.Namespace.Name).List(metav1.ListOptions{
+		LabelSelector: podLabelSelector,
+	})
+	Expect(err).NotTo(HaveOccurred())
+	if len(pods.Items) != 1 {
+		Expect(fmt.Errorf("expected single kube-oidc-proxy pod running, got=%d", len(pods.Items))).NotTo(HaveOccurred())
+	}
+
+	tmpDir, err := ioutil.TempDir(os.TempDir(), "kube-oidc-proxy-e2e")
+	Expect(err).NotTo(HaveOccurred())
+
+	defer func() {
+		err := os.RemoveAll(tmpDir)
+		Expect(err).NotTo(HaveOccurred())
+	}()
+
+	logFile := filepath.Join(tmpDir, "log.txt")
+
+	err = f.Helper().Kubectl(f.Namespace.Name).Run(
+		"cp", fmt.Sprintf("%s:audit-log", pods.Items[0].Name), logFile)
```

Comment:
`kubectl cp` does not exist in newer versions of Kubernetes as far as I'm aware? and/or it is being removed due to the CVEs discovered a while ago. Not sure on the latest news on that...

Reply:
Need to think of something to replace this with then

Reply:
:+1: given this is only in the test cases, I think we can merge this (as the test will start failing anyway if/when `kubectl` removes this feature)